### PR TITLE
Regenerate docs (if necessary) during cherry-pick operations.

### LINKED ISSUE #44533: manpage artifacts are not being generated in a release
**Kubernetes version** (use `kubectl version`): 1.6.1

**What happened**:
I downloaded https://dl.k8s.io/v1.6.1/kubernetes.tar.gz
The files `doc/man/man1/kube-apiserver.1`and others just contains the 'this is an autogenerated doc place holder`

The markdown docs like in `doc/admin/kube-apiserver.md` etc. also just contains the placeholder text.

**What you expected to happen**:
I'm expecting the release tarball to contain the generated manpages similar to the markdown ones in https://github.com/kubernetes/kubernetes.github.io/blob/release-1.6/docs/admin/kube-apiserver.md

### PR DESCRIPTION
**Pull request by david-mcmahon**
This change ensures that cherrypicks are coupled with any related doc generation required.
closes #44533

### CODE REVIEW

**Comment by fejta** [comment]
LGTM
/unassign
/assign @pwittrock

**Comment by david-mcmahon** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by david-mcmahon** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by david-mcmahon** [comment]
@kubernetes/test-infra-maintainers what am I missing here?  The logs seem to indicate a flake. I can't imagine what from this change is causing any test failures.

**Comment by krzyzacy** [comment]
maybe https://github.com/kubernetes/kubernetes/issues/47135

**Comment by marun** [comment]
@pwittrock Does this belong in the v1.7 milestone?

**Comment by pwittrock** [comment]
Looks good, but I don't know much about generate docs state, or if we even still need it.

File: hack/cherry_pick_pull.sh [hunk 1]
```diff
@@ -183,6 +183,15 @@ for pull in "${PULLS[@]}"; do
 done
 gitamcleanup=false
 
+# Re-generate docs (if needed)
+echo
+echo "Regenerating docs..."
+if ! hack/generate-docs.sh; then
+  echo
+  echo "hack/generate-docs.sh FAILED to complete."
+  exit 1
```
Comment (liggitt) [inline hack/cherry_pick_pull.sh]: this also breaks cherry-picks to 1.6 and leaves them with lots of uncommitted generated docs in the working dir

**Pull request merged**